ohpc-build.yaml

```diff
@@ -8,8 +8,8 @@
     - { name: 'ohpc_jupyter', tags: 'ohpc_jupyter', when: jupyter_provision}
     - { name: 'ohpc_matlab', tags: 'ohpc_matlab', when: matlab_provision }
     - { name: 'ohpc_sas', tags: 'ohpc_sas', when: sas_provision }
-    - { name: 'ohpc_rabbitmq', tags: 'ohpc_rabbitmq', when: rabbitmq_provision }
+    - { name: 'ohpc_rabbitmq', tags: 'ohpc_rabbitmq', when: enable_user_reg }
```

Comment:
I think we should still have `rabbitmq_provision` flag here because RabbitMQ installation can be a dependency for other apps too besides user_reg app. For example, if we had another application that should use RabbitMQ and we turn off the enable_user_reg flag, then we don't have rabbitMQ installed on the master. 

I agree with tying the agents to `enable_user_reg` flag like in line 15 because they are user_reg app dependencies, but RabbitMQ install though is a dependency, it needs to be have a flag on its own for future apps that may use it. I t preserves modularity between apps going forward.

Reply:
I agree that this is more flexible.   We should set the rabbitmq_provision flag to True by default to accomplish the same.  Ideally this flag would become an OR conditional of all subsystems that depend on rabbitmq so that it is installed whenever any one of those subsystems is installed.